replica/replica.go

```diff
@@ -104,14 +105,18 @@ func (r *Replica) Stop() error {
   return nil
 }
 
-func NewReplica(db ethdb.Database, config *eth.Config, ctx *node.ServiceContext, transactionProducer TransactionProducer, consumer cdc.LogConsumer) (*Replica, error) {
+func NewReplica(db ethdb.Database, config *eth.Config, ctx *node.ServiceContext, transactionProducer TransactionProducer, consumer cdc.LogConsumer, syncShutdown bool) (*Replica, error) {
   go func() {
     for operation := range consumer.Messages() {
       operation.Apply(db)
     }
   }()
   <-consumer.Ready()
   log.Info("Replica up to date")
+  if syncShutdown {
+    log.Info("Replica shutdown after sync flag was set, shutting down")
+    os.Exit(0)
```

Comment:
I could see implementing the `Stop` function that currently just `return nil` online 104, instead of calling `os.Exit(0)`

Reply:
We may revisit this in the future, but I think it's okay for now.